controller/rest/rest.go

```diff
@@ -1425,6 +1435,16 @@ func InitContext(ctx *Context) {
 	if err := initJWTSignKey(); err != nil {
 		log.WithError(err).Error("failed to initialize JWT sign key.")
 	}
+
+	searchRegistries = utils.NewSet()
+
+	for _, reg := range strings.Split(ctx.SearchRegistries, ",") {
+		k := fmt.Sprintf("https://%s/", strings.Trim(reg, " "))
```

Comment:
Good point, specifying registries as a full URL currently would result in unpredictable behavior. We assumed only the hostname is provided. `url.Parse` will not give the desired results in this case.

I don't think we should introduce the complexity of supporting multiple notations for this parameter, but instead be clear about what format the URLs should be provided in.

I'll update the usage of the flag to be more clear about the expected input. Should I also update this section, so that registries can be provided as a full URL, or do we stick with hostnames?

Reply:
We added a `url.Parse` with a fallback to the registry when no `Host` could be parsed. This allows specifying both a hostname and a full URL, and it also provides an error message when the registry could not be parsed.